pyfstat/utils/converting.py

```diff
@@ -107,3 +107,62 @@ def gps_to_datestr_utc(gps):
         tzinfo=timezone.utc,
     )
     return dt.strftime("%c %Z")
+
+
+def convert_h0_cosi_to_aCross_aPlus(h0, cosi):
+    """
+    Converts amplitude parameters from a pair of `(h0,cosi)` to a pair of `(aPlus,aCross)`.
+
+    Parameters
+    -------
+    h0: float
+        Nominal GW amplitude.
+    cosi: float
+        Cosine of the source inclination w.r.t. line of sight.
+
+    Returns
+    -------
+    aPlus: float
+        Plus polarization amplitude.
+    aCross: float
+        Cross polarization amplitude.
+    """
+    aPlus = 0.5 * h0 * (1 + cosi**2)
+    aCross = h0 * cosi
+    return aPlus, aCross
+
+
+def convert_aCross_aPlus_to_h0_cosi(aPlus, aCross):
+    """
+    Converts amplitude parameters from a pair of `(aPlus,aCross)` to a pair of `(h0,cosi)`.
+
+    Inverse to ``convert_h0_cosi_to_aCross_aPlus()``.
+
+    Conversion in this direction is only well-defined if `abs(aCross) > abs(aPlus)`,
+    as expected for GWs from neutron stars at twice the spin frequency,
+    but not necessarily in all other CW emission scenarios.
+
+    Parameters
+    -------
+    aPlus: float
+        Plus polarization amplitude.
+    aCross: float
+        Cross polarization amplitude.
+
+    Returns
+    -------
+    h0: float
+        Nominal GW amplitude.
+    cosi: float
+        Cosine of the source inclination w.r.t. line of sight.
+    """
+
+    if np.abs(aCross) > np.abs(aPlus):
+        raise ValueError("not valid for abs(aCross)>abs(aPlus)")
+
+    h0 = aPlus + np.sqrt(aPlus**2 - aCross**2)
+    if h0 > 0:
+        cosi = aCross / h0
+    else:
+        cosi = 0
```

Comment:
@dbkeitel what am I missing here? `h0 > 0` by definition?

```suggestion
    cosi = aCross / h0
```

Reply:
No, only `h0 >= 0` is guaranteed. ;)

Turning around the if-else, so the special case is specifically labelled `==0`, would make it more intuitive, but this order is inherited from CFSv2.

Reply:
```suggestion
    cosi = np.divide(aCross, h0, out=np.zeros_like(aCross), where=(h0 > 0))
```